### PR TITLE
Allow intervals less than 60 seconds

### PR DESCRIPTION
**Pull request by daryllstrauss**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [x] adds new tests (if appropriate)
- [x] updates `CHANGELOG.md` (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?
IntervalClock should allow any interval greater than 0.

## Why is this PR important?
60 seconds was an arbitrary time period. Some applications can use short intervals.

### CODE REVIEW

File: src/prefect/schedules/clocks.py [hunk 1]
```diff
@@ -74,8 +74,8 @@ def __init__(
     ):
         if not isinstance(interval, timedelta):
             raise TypeError("Interval must be a timedelta.")
-        elif interval.total_seconds() < 60:
-            raise ValueError("Interval can not be less than one minute.")
+        elif interval.total_seconds() <= 0:
+            raise ValueError("Interval most be greater than 0.")
```
Comment (jlowin) [inline src/prefect/schedules/clocks.py]: Typo: “must”

**Comment by daryllstrauss** [comment]
Fixed typo and updated CHANGELOG.md

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ These changes are available in the [master branch](https://github.com/PrefectHQ/
 
 - Allow for task looping to beyond Python's maximum recursion depth - [#1862](https://github.com/PrefectHQ/prefect/pull/1862)
 - Prevent duplication of stdout logs from multiple instantiated agents - [#1866](https://github.com/PrefectHQ/prefect/pull/1866)
+- Allow intervals less than 60 seconds - [#1880](https://github.com/PrefectHQ/prefect/pull/1880)
```
Comment (cicdw) [inline CHANGELOG.md]:
```suggestion
- Allow intervals less than 60 seconds in `IntervalClock`s - [#1880](https://github.com/PrefectHQ/prefect/pull/1880)
```
to clarify, as `CronClock`s for example can never go sub-minute

File: CHANGELOG.md [hunk 3]
```diff
@@ -31,7 +32,7 @@ These changes are available in the [master branch](https://github.com/PrefectHQ/
 
 ### Contributors
 
-- None
+- Daryll Strauss <daryll.strauss@gmail.com>
```
Comment (cicdw) [inline CHANGELOG.md]:
```suggestion
- [Daryll Strauss](daryll.strauss@gmail.com)
```

**Pull request merged**